api/helpers/game-states/moves/seven-scuttle/validate.js

```diff
@@ -0,0 +1,71 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Validate request to seven-scuttle',
+
+  description: 'Verifies whether a request to scuttle with a seven is legal, throwing an explanatory error if not.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'Object containing the current game state',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - Object describing the request to scuttle
+     * @param { String } requestedMove.cardId - Card Played for scuttle
+     * @param { String } requestedMove.targetId - Card Targeted for scuttle
+     * @param { MoveType.SEVEN_SCUTTLE } requestedMove.moveType - Specifies that this a scuttle
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'Object describing the request to scuttle',
+      required: true,
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting the move',
+      required: true,
+    },
+  },
+
+  sync: true,
+
+  fn: ({ requestedMove, currentState, playedBy }, exits) => {
+    try {
+      const player = playedBy ? currentState.p1 : currentState.p0;
+      const opponent = playedBy ? currentState.p0 : currentState.p1;
+
+      const playedCard = player.hand.find(({ id }) => id === requestedMove.cardId);
```

Comment:
Because we are resolving a seven, the played card should be taken from the top two cards of the deck, not the player's hand. Playing a seven lets you pick one of the top two cards from the deck to play any way you choose. So all the different sevenSomething move typed e.g. sevenScuttle are finishing your seven effect by playing a card from the top of the deck.

That means the card used to scuttle (the playedCard) must come from the top of the deck rather than the player's hand. It also means that this effect is only allowed in the situation where we are in the middle of resolving a seven (meaning the player just played a seven that their opponent declined to counter and the player hasn't yet chosen which way they will play a card from the top of the deck).

All this means we need an additional check to confirm that the gamephase is `GamePhase.RESOLVING_SEVEN` and that when we capture the playedCard that we should do so from the top two cards of the deck instead of the player hand. Check out the existing `api/helpers/gameStates/moves/seven-points/validate.js` helper for examples of both of these